signale.js

```diff
@@ -78,11 +78,11 @@ class Signale {
   }
 
   _logger(type, ...messageObj) {
-    this._log(this._buildSignale(this._types[type], ...messageObj));
+    this._log(this._buildSignale(this._types[type], ...messageObj), this._types[type].stream);
   }
 
-  _log(message) {
-    this._stream.write(message + '\n');
+  _log(message, stream = this._stream) {
+    stream.write(message + '\n');
```

Comment:
Nice idea! I can see how this might be useful, but we can apply it a way that supports multiple writable streams, in order to also close #26. The user will be able to define an array of valid writable streams under the `stream` logger attribute, which will override, for only that logger, both the default `process.stdout` as well as any other user-defined stream used when initializing a new `Signale` instance e.g. 

```js
const options = {
  stream: [stream_1, stream_2], // Overrides the default `process.stdout` with `stream_1` & `stream_2` for all loggers
  types: {
    error: {
     stream: [stream_3, stream_4] // Overrides both the default `process.stdout` and `stream_1` & `stream_2` with `stream_3` & `stream_4` only for the .error() logger
    }
  }
}
const signale = new Signale(options);
signale.error('Foo'); // 'Foo' will appear only on `stream_3` & `stream_4`
signale.success('Bar'); // 'Bar' will appear only on `stream_1` & `stream_2`
```

A new `  _formatStream(stream)` method can be created on line 87, just above `_formatDate()`, that will make sure that the user-defined streams are packed into an array:

```js
_formatStream(stream) {
  return Array.isArray(stream) ? stream : [stream];
}
```

And then we can write on all streams like so:

```js
_log(message, streams = this._stream) {
  this._formatStream(streams).forEach(stream => {
    stream.write(message + '\n');
  });
}
```

Reply:
Thanks for the feedback. I agree it would be an easy win to fix #26 too!

I've done the work requested however I didn't extract the array type consolidation to a function as I couldn't find anywhere that used it and it didn't feel like it fit in the same vein as the other format* functions which were string formatting. Happy to change this if you want though.

Reply:
A separate format function would be useful if access to writable streams was required on more than one different parts of the code and also for keeping things as modular as possible, but it is a change easily applied if needed in the future. Also, it would be better if the readme update could go to a separate pull request on its own, so it can be easily merged when ready to release  the new version. Everything else looks awesome! Thank you a lot for taking the time to contribute : )

Reply:
Yep, I believe fast failure is better since it is up to the user to provide the valid writable streams and resolve any possible issue at first sight. Just a final tweak [here](https://github.com/klauscfhq/signale/pull/27/files#r191817012) and it is good to be merged. Also, thank you a lot for your readme PR!